pkg/actuators/machine/machine_scope.go

```diff
@@ -0,0 +1,164 @@
+package machine
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	machinev1 "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	machineapierros "github.com/openshift/machine-api-operator/pkg/controller/machine"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+	awsproviderv1 "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsprovider/v1beta1"
+	awsclient "sigs.k8s.io/cluster-api-provider-aws/pkg/client"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	userDataSecretKey = "userData"
+)
+
+// machineScopeParams defines the input parameters used to create a new MachineScope.
+type machineScopeParams struct {
+	context.Context
+
+	// client reader that bypasses the manager's cache
+	apiReader runtimeclient.Reader
+	// api server controller runtime client
+	client runtimeclient.Client
+	// machine resource
+	machine *machinev1.Machine
+}
+
+type machineScope struct {
+	context.Context
+
+	// client for interacting with AWS
+	awsClient awsclient.Client
+	// client reader that bypasses the manager's cache
+	apiReader runtimeclient.Reader
```

Comment:
I might have missed it, but I'm not sure if this is used anywhere, might be worth getting rid of it if it isn't used? WDYT?